src/NexusMods.DataModel/ArchiveInstaller.cs

```diff
@@ -218,6 +214,7 @@ private async Task SetStatus(ModId modId, ModStatus status)
         _logger.LogInformation("Setting status of ModId:{ModId}({Name}) to {Status}", modId, mod.Name, status);
         
         using var tx = _conn.BeginTransaction();
+        tx.Add(modId.Value, Mod.Name, $"Installation failed: {mod.Name}");
```

Comment:
This method was meant to support setting any status, not just failed status.
Either change the method to be SetFailedStatus, or check if status==ModStatus.Failed before replacing the name